os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,201 @@
+# IBM Z Secure Execution
+---
+
+# Theory of operation
+
+[IBM Secure Execution for Linux®](https://www.ibm.com/docs/en/linux-on-systems?topic=virtualization-introducing-secure-execution-linux) is a hardware-based security technology that is
+built into the IBM z15™ and LinuxONE III generation systems. It is designed to
+provide scalable isolation for individual workloads to help protect them from not
+only external attacks, but also insider threats. In traditional x86 ring architectures,
+the host can access the memory and data of guest applications freely, leading to the
+potential for malicious software to be proliferated throughout the entire system.
+Isolation between host and guest environments is necessary to help prevent system compromise.
+Users can enable Secure Execution feature on IBM zKVM Host and Guest VMs to isolate them from each other.
+
+That means:
+1. An encrypted Linux image, called `sd-boot`, (kernel + ramdisk + cmdline) is created using the host public key and a customer-specific key. Since the encryption keys are stored on the IBM Z hardware and firmware, the encrypted image can only be executed in a virtual machine on the host(s) it has been prepared for, and the image can’t be decrypted or tampered with outside of the designated host(s).
+2. zKVM host cannot access memory of secured guest VM
+3. Guest VMs cannot access memory of other secured guest VMs
+4. If anyone tries to access memory of secured guest VM, than this gets blocked on HW level
+
+## Specific attacks being addressed
+- protect guests against the system
+- protect guests against a Linux Admin
+- protect services in KVM guests against each other
+
+## Specific attacks not being addressed
```

Comment:
Okay, so as I understand this:
- We're not securing provisioning, so we're okay with the host accessing/modifying guest memory on first boot and potentially installing a rootkit
- We're not requiring disk encryption, so we're okay with the host reading guest data
- We're not requiring authenticated disk encryption, so we're okay with the host compromising software on the guest's disk

In what way are we protecting guests against the system, a Linux admin, or each other?

Reply:
as i understand it - main goal is to protect guest's RAM, disk encryption is a bit aside (but recommended)
@hbrueckner , @holgwolf wdyt?